Library/Homebrew/download_strategy.rb

```diff
@@ -608,6 +608,59 @@ def fetch_release_metadata
   end
 end
 
+class ScpDownloadStrategy < AbstractFileDownloadStrategy
+  attr_reader :tarball_path, :temporary_path
+
+  def initialize(name, resource)
+    super
+    @tarball_path = HOMEBREW_CACHE/"#{name}-#{version}#{ext}"
+    @temporary_path = Pathname.new("#{cached_location}.incomplete")
+    parse_url_pattern
+  end
+
+  def parse_url_pattern
+    url_pattern = %r{scp://([^@]+@)?([^@:/]+)(:[0-9]+)?/(\S+)}
+    unless @url =~ url_pattern
+      raise CurlDownloadStrategyError, "Invalid URL for scp: #{@url}"
+    end
+
+    _, @user, @host, @port, @path = *@url.match(url_pattern)
+  end
+
+  def fetch
+    ohai "Downloading #{@url}"
+
+    if cached_location.exist?
+      puts "Already downloaded: #{cached_location}"
+    else
+      begin
+        safe_system "scp", scp_source, temporary_path.to_s
+      rescue ErrorDuringExecution
+        raise CurlDownloadStrategyError, @url
+      end
+
+      ignore_interrupts { temporary_path.rename(cached_location) }
+    end
+  end
+
+  def cached_location
+    tarball_path
+  end
+
+  def clear_cache
+    super
+    rm_rf(temporary_path)
+  end
+
+  private
+
+  def scp_source
+    path_prefix = @path.start_with?("~") ? "" : "/"
```

Comment:
`path_prefix = "/" if @path.start_with?("~")`

Reply:
Nifty—never thought of doing it that way.

Reply:
Not sure if this changes your guidance, but I realized this would be `path_prefix = "/" unless @path.start_with("~")`